lambda_code/tests/test_fhir_repository.py

```diff
@@ -188,3 +249,75 @@ def test_delete_throws_error_when_response_can_not_be_handled(self):
 
         # Then
         self.assertDictEqual(e.exception.response, response)
+
+
+class TestFindImmunizations(unittest.TestCase):
+    def setUp(self):
+        self.table = MagicMock()
+        self.repository = ImmunisationRepository(table=self.table)
+
+    def test_find_immunizations(self):
+        """it should find events with nhsNumber and diseaseCode(like snomed)"""
+        nhs_number = "a-patient-id"
+        disease_code = "a-snomed-code-for-disease"
+        dynamo_response = {"ResponseMetadata": {"HTTPStatusCode": 200}, "Items": []}
+        self.table.query = MagicMock(return_value=dynamo_response)
+
+        condition = Key("PatientPK").eq(_make_patient_pk(nhs_number))
+        sort_key = f"{disease_code}#"
+        condition &= Key("PatientSK").begins_with(sort_key)
+
+        # When
+        _ = self.repository.find_immunizations(nhs_number, disease_code)
+
+        # Then
+        self.table.query.assert_called_once_with(
+            IndexName="PatientGSI",
+            KeyConditionExpression=condition,
+            FilterExpression=ANY,
+        )
+
+    def test_exclude_deleted(self):
+        """it should exclude records with DeletedAt attribute"""
+        dynamo_response = {"ResponseMetadata": {"HTTPStatusCode": 200}, "Items": []}
+        self.table.query = MagicMock(return_value=dynamo_response)
+
+        is_ = Attr("DeletedAt").not_exists()
+
+        # When
+        _ = self.repository.find_immunizations("an-id", "a-code")
+
+        # Then
+        self.table.query.assert_called_once_with(
+            IndexName="PatientGSI",
+            KeyConditionExpression=ANY,
+            FilterExpression=is_,
+        )
+
+    def test_map_results_to_immunizations(self):
+        """it should map Resource list into a list of Immunizations"""
+        imms1 = {"id": 1}
+        imms2 = {"id": 2}
+        items = [{"Resource": json.dumps(imms1)}, {"Resource": json.dumps(imms2)}]
+
+        dynamo_response = {"ResponseMetadata": {"HTTPStatusCode": 200}, "Items": items}
+        self.table.query = MagicMock(return_value=dynamo_response)
+
+        # When
+        results = self.repository.find_immunizations("an-id", "a-code")
+
+        # Then
+        self.assertListEqual(results, [imms1, imms2])
+
+    def test_bad_response_from_dynamo(self):
+        """it should throw UnhandledResponse when the response from dynamodb can't be handled"""
+        bad_request = 400
```

Comment:
Should this be 500?

Reply:
Any error response from dynamodb is an UnhandledResponse for us. So, in this case it doesn't make any difference.